scripts/base/frameworks/analyzer/main.zeek

```diff
@@ -232,16 +237,25 @@ function disable_analyzer(tag: AllAnalyzers::Tag) : bool
 	return __disable_analyzer(tag);
 	}
 
-function register_for_ports(tag: Analyzer::Tag, ports: set[port]) : bool
+function register_for_ports(tag: Analyzer::Tag, server_ports: set[port], non_server_ports: set[port] &default=set()) : bool
 	{
 	local rc = T;
 
-	for ( p in ports )
+	for ( p in server_ports )
 		{
 		if ( ! register_for_port(tag, p) )
 			rc = F;
 		}
 
+	for ( p in non_server_ports )
+		{
+		if ( ! register_for_port(tag, p) )
+			rc = F;
+		}
+
+	# Automatically update likely_server_ports with server_ports
+	likely_server_ports += server_ports;
```

Comment:
Interesting, I think I never realized that `const &redef` would allow you to do that, but i guess that makes some sense, but it also breaks my mental model about anything `const &redef` not changing at runtime.

So, technically this needs a new BiF to invalidate the cache or better transport the new likely_server_ports "into" the core.

https://github.com/zeek/zeek/blob/b2488808ffd1eca98412e96a7fcabdf5e71a8ef2/src/packet_analysis/protocol/ip/IPBasedAnalyzer.cc#L144-L155

But can certainly do as a follow-up.